Add Backup Documentation (#2830)
kyma-project/kyma: docs/backup/docs/03-01-backup-configuration.md

---
title: Configuration
---

> The configuration of ark consist of two custom resources: 

Reviewer: ```suggestion
Ark configuration consists of two custom resources: 
```

- The VolumeSnapshotLocation CR defines the provider of physical volume snapshots.
- The BackupStorageLocations CR defines a bucket or storage location for cluster resources.

> A kyma installation is delivered with a set of default snapshot and storage locations. If needed it is possible to add custom locations in the `heptio-ark` namespace.

Reviewer: ```suggestion
A Kyma installation provides a set of default snapshot and storage locations. If necessary, you can add custom locations in the `heptio-ark` Namespace.
```

> A sample YAML `BackupStorageLocation` configuration file looks as follows:

Reviewer: ```suggestion
A sample `BackupStorageLocation` configuration file looks as follows:
```

```apiVersion: ark.heptio.com/v1
kind: BackupStorageLocation
metadata:
  name: default
  namespace: heptio-ark
spec:
  config:
    resourceGroup: BackupStorage
    storageAccount: foo...
  objectStorage:
    bucket: bucket
  provider: azure
```

For a detailed description of all options, see the [Heptio Ark documentation](https://github.com/heptio/velero/blob/master/docs/api-types/backupstoragelocation.md).

A sample `VolumeSnapshotLocation` configuration file looks as follows:

```apiVersion: ark.heptio.com/v1
kind: VolumeSnapshotLocation
metadata:
  name: azure-default
  namespace: heptio-ark
spec:
  config:
    apiTimeout: 15m
  provider: azure
```

> A detailed description of all options can be found in the [Heptio Ark Documentation](https://github.com/heptio/velero/blob/master/docs/api-types/volumesnapshotlocation.md).

Reviewer: ```suggestion
For a detailed description of all options, see the [Heptio Ark Documentation](https://github.com/heptio/velero/blob/master/docs/api-types/volumesnapshotlocation.md).
```
